src/Components/Main.js

```diff
@@ -26,35 +24,44 @@ function Main() {
       ...tasks,
       newTask,
     ];
-
-    setTasks(newTasks);
```

Comment:
You need to send to the server here, and then either `reload()` or `setTasks()`.